compile_data.py

```diff
@@ -79,6 +80,17 @@ def get_evals_json(src_data_dir):
     df['hrs'] = df['hrs'].astype(float)
     return df
 
+def wavg(group, avg_name, weight_name):
```

Comment:
The point of keeping only the last observation was that course syllabi change every year and so the last year's record is the most reflective. 
It probably makes sense to take a weighted average of the last few observations (3?), weighting them inversely proportional to their age.

UPD: my bad, I commented on this before reading the `recentsDict` part below.  Please ignore the comment above.